File: app/src/main/java/com/pixels/blockies/app/game/Block.java
package com.pixels.blockies.app.game;

import com.pixels.blockies.app.game.figures.Rotatable;

public class Block {
    int x = -1;
    int y = -1;
    Rotatable figure = null;

    public Block(Rotatable figure)
    {
        this.figure = figure;
    }

    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y;
    }

    public int getInner(int x, int y) {
        return figure.get()[y][x];
    }

    public int getOffsetY() {
        return figure.get().length;
    }

    public int getOffsetX() {
        return figure.get()[0].length;
    }

    public void rotate(){
        figure.rotate();
    }

    public int getRotatedInner(int x, int y){
        return figure.getNext()[y][x];
    }

    public int getRotatedOffsetY() {
        return figure.getNext().length;
    }

    public int getRotatedOffsetX() {
        return figure.getNext()[0].length;
    }

}

File: app/src/main/java/com/pixels/blockies/game/game/Block.kt
package com.pixels.blockies.game.game

import com.pixels.blockies.game.game.figures.Rotatable

/**
 * Represents a logical block with a coordinate in the grid
 * and a figure that can be rotated
 */
class Block(figure: Rotatable?) {
    var x = -1
    private var y = -1
    private var figure: Rotatable? = null

    init {
        this.figure = figure
    }

    fun getY(): Int {
        return y
    }

    fun setY(y: Int) {
        this.y = y
    }

    fun getInner(x: Int, y: Int): Int {
        return figure!!.get()[y]!![x]
    }

    fun getOffsetY(): Int {
        return figure!!.get().size
    }

    fun getOffsetX(): Int {
        return figure!!.get()[0]!!.size
    }

    fun rotate() {
        figure!!.rotate()
    }

    fun getRotatedInner(x: Int, y: Int): Int {
        return figure!!.getNext()[y]!![x]
    }

    fun getRotatedOffsetY(): Int {
        return figure!!.getNext().size
    }

    fun getRotatedOffsetX(): Int {
        return figure!!.getNext()[0]!!.size
    }
}